packages/fxa-auth-server/lib/routes/account.js

```diff
@@ -1490,6 +1490,47 @@ module.exports = (
         return { subscriptions };
       },
     },
+    {
+      method: 'PUT',
+      path: '/account/ecosystemAnonId',
+      apidoc: {
+        errors: [error.invalidScopes, error.anonIdExists],
+      },
+      options: {
+        auth: {
+          payload: false,
+          strategy: 'oauthToken',
+        },
+        validate: {
+          payload: {
+            ecosystemAnonId: isA.string().required(),
+          },
+        },
+      },
+      handler: async function (request) {
+        log.begin('account.updateEcosystemAnonId', request);
+
+        const { uid, scope } = request.auth.credentials;
```

Comment:
Because this is doing a DB update, I suspect we will need some kind of rate limiting here. Something like

```
await customs.checkAuthenticated(request, uid, 'updateEcosystemAnonId');
```

The `checkAuthenticated` call allows for [100 requests](https://github.com/mozilla/fxa/blob/77df83c4500695cef143b0b8850dd2d6215c712a/packages/fxa-customs-server/lib/config/config.js#L166) in 15mins per uid before rate limited.